src/history.cpp

```diff
@@ -170,4 +171,60 @@ namespace mamba
         // }
         return map;
     }
+
+    void History::add_entry(const std::vector<History::UserRequest> user_request)
+    {
+        std::ofstream out;
+        out.open(m_history_file_path, std::ios_base::app);
+        if (out.fail())
+        {
+            throw std::runtime_error("Couldn't open file: " + m_history_file_path.string());
+        }
+        else
+        {
+            for (auto request : user_request)
+            {
+                out << "==> " << request.date << " <==" << std::endl;
+                out << "# cmd: " << request.cmd << std::endl;
+                out << "# conda version: " << request.conda_version << std::endl;
+                for (auto unlink_dist : request.unlink_dists)
+                {
+                    out << unlink_dist << std::endl;
+                }
+                for (auto link_dist : request.link_dists)
+                {
+                    out << link_dist << std::endl;
+                }
+                if (request.update.size() > 0)
+                {
+                    out << "# update specs: [";
+                    std::string update_string;
+                    for (auto u : request.update)
+                    {
+                        update_string += "'" + u + "',";
+                    }
+                        update_string.back() = ']';
+                        out << update_string << std::endl;
+                }
+                if (request.remove.size() > 0)
+                {
+                    out << "# remove specs: [";
+                    for (auto r : request.remove)
+                    {
+                        out << "'" << r << "',";
+                    }
+                        out << "]" << std::endl;
+                }
+                if (request.neutered.size() > 0)
```

Comment:
Neutered and removed needs an update string logic, otherwise it will format things wrongly.
Forgot about this.